recipes/cgltf/all/conanfile.py

```diff
@@ -1,89 +1,88 @@
-            "#define CGLTF_WRITE_IMPLEMENTATION\n"
-            "#include \"cgltf_write.h\"\n"
-        )
-        tools.save(os.path.join(self.build_folder, self._source_subfolder, "cgltf.c"), cgltf_c)
-        tools.save(os.path.join(self.build_folder, self._source_subfolder, "cgltf_write.c"), cgltf_write_c)
+        cgltf_c = '#define CGLTF_IMPLEMENTATION\n#include "cgltf.h"\n'
+        cgltf_write_c = '#define CGLTF_WRITE_IMPLEMENTATION\n#include "cgltf_write.h"\n'
+        save(self, os.path.join(self.build_folder, self.source_folder, "cgltf.c"), cgltf_c)
+        save(self, os.path.join(self.build_folder, self.source_folder, "cgltf_write.c"), cgltf_write_c)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.configure()
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
 
     def build(self):
         self._create_source_files()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=self.source_path.parent)
         cmake.build()
 
-    def package(self):
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
-        cmake.install()
-        for header_file in ["cgltf.h", "cgltf_write.h"]:
-            header_fullpath = os.path.join(self.package_folder, "include", header_file)
-            self._remove_implementation(header_fullpath)
-
-    @staticmethod
-    def _remove_implementation(header_fullpath):
-        header_content = tools.load(header_fullpath)
+    def _remove_implementation(self, header_fullpath):
+        header_content = load(self, header_fullpath)
         begin = header_content.find("/*\n *\n * Stop now, if you are only interested in the API.")
         end = header_content.find("/* cgltf is distributed under MIT license:", begin)
         implementation = header_content[begin:end]
-        tools.replace_in_file(
+        replace_in_file(
+            self,
             header_fullpath,
             implementation,
-            (
-                "/**\n"
-                " * Implementation removed by conan during packaging.\n"
-                " * Don't forget to link libs provided in this package.\n"
-                " */\n\n"
-            )
+            "/**\n * Implementation removed by conan during packaging.\n * Don't forget to link libs provided in this package.\n */\n\n",
         )
 
+    def package(self):
+        copy(self, "LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+        for header_file in ["cgltf.h", "cgltf_write.h"]:
+            header_fullpath = os.path.join(self.package_folder, "include", header_file)
+            self._remove_implementation(header_fullpath)
+        for dll in (self.package_path / "lib").glob("*.dll"):
```

Comment:
You probably want to change this as you did for the other PRs

Reply:
For this one it was actually needed. Here's a CI run without it:
https://github.com/conan-io/conan-center-index/pull/18732#issuecomment-1746553400

The CMakeLists.txt should be tweaked to correctly handle this during the build, but simply copying the DLLs is sufficient. It at least avoids the need for additional patches.

Reply:
Thanks for the extra insight then! :)